### PR TITLE
feat: Offline Support with nested schema

### PR DESCRIPTION
**Pull request by Lavanyagaur22**
### Description
Inside the `interceptAsync()` of `ConflictInterceptor` class, I have checked for the network connection. If the network connection is not there, and if the request operation is a mutation, I store them in an array list as of now. 
Whenever the net comes, we wait for a mutation request from the user side. Upon getting it, we fetch the mutation requests stored in the list and replicate them back to the server. 
We are not providing UI bindings to the user currently.

### How to test it?
- Start server.
- Go offline. 
- Make some mutations. (For updating a task, click on the task and provide the details in it. )
- Come online.
- Make any mutation. And you will see the results of the mutations done when offline along with the result of the mutation done when the network connection is regained.

**Also trying to study request so that we can parse it accordingly and store important things in database and get rid of array list**

### CODE REVIEW

File: offix/src/main/java/org/aerogear/offix/OfflineCallback.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import android.util.Log
+import com.apollographql.apollo.exception.ApolloException
+import com.apollographql.apollo.interceptor.ApolloInterceptor
+
+class OfflineCallback(val request: ApolloInterceptor.InterceptorRequest) : ApolloInterceptor.CallBack {
+    val TAG = javaClass.simpleName
+    override fun onResponse(response: ApolloInterceptor.InterceptorResponse) {
+        Log.d(TAG, "onResponse()")
+//
+//        /* Check if the conflict is present in the response of not using the ConflictResolutionHandler class.
```
Comment (wtrocki) [inline offix/src/main/java/org/aerogear/offix/OfflineCallback.kt]: ??

**Comment by wtrocki** [comment]
> We are not providing UI bindings to the user currently.

That is absolutely fine. How to test this PR?

**Comment by Lavanyagaur22** [comment]
@wtrocki 

> > We are not providing UI bindings to the user currently.
> 
> That is absolutely fine. How to test this PR?

### How to test it?
- Start server.
- Go offline.
- Make some mutations. (For updating a task, click on the task and provide the details in it. )
- Come online.
- Make any mutation. And you will see the results of the mutations done when offline along with the result of the mutation done when the network connection is regained.

**Comment by harshithdwivedi** [comment]
@Lavanyagaur22 is this done? 
If so, we can drop the WIP tag maybe :)

**Review by harshithdwivedi — APPROVED** [review]
LGTM, good work.
P.S. Maybe go a little easy with the logs the next time hehe

**Comment by wtrocki** [comment]
Went offline. Created couple items, edited them. Went online - Do see updates failed on server.

Second test. Created item online. Went offline done two edits. Back online.  Got conflict. 

Missing anything?

**Comment by wtrocki** [comment]
I think I have mixed things up. Does offline create work for you?

**Comment by Lavanyagaur22** [comment]
## 1. When offline:
- Updated 2 tasks: (Task 1 and Task 2)
- Then came online, and updated Task 3

### Result of the 3 mutation requests: 
Screenshot 2019-08-16 at 12 53 16 PM

## 2. When offline:
@wtrocki Create Task is working fine.

**Comment by Lavanyagaur22** [comment]
> I think I have mixed things up. Does offline create work for you?

Yes, it's working. Maybe you can try it now once again, you might be ending up in null ptr exception because of the null in version field which might be resulting in the crashing of your app. Fixed it now.

**Comment by Lavanyagaur22** [comment]
> I think I have mixed things up. Does offline create work for you?

@wtrocki Yes, it's working. Maybe you can try it now once again, you might be ending up in null ptr exception because of the null in version field which might be resulting in the crashing of your app. Fixed it.

**Comment by Lavanyagaur22** [comment]
Also, @harshithdwivedi @wtrocki Our offline requests are working perfect and the results can be seen on the server too. But when the offline create mutations hit the server successfully, changes are not being reflected back in Apollo cache.

**Comment by wtrocki** [comment]
Ok. So lack of item in cache can explain it. Do you think we can have subscriptions implemented to get updates to cache for now?

**Comment by Lavanyagaur22** [comment]
> Ok. So lack of item in cache can explain it. Do you think we can have subscriptions implemented to get updates to cache for now?

Yeah, I was about to open issue for using subscriptions in app like we were using earlier.

**Comment by wtrocki** [comment]
Let's address Codacy issues and merge.

**Review by harshithdwivedi — APPROVED** [review]

**Pull request merged**